core/ui-session/src/web3Helper.ts

```diff
@@ -0,0 +1,74 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+import { config } from '@evan.network/ui-dapp-browser';
+import { Web3 } from '@evan.network/api-blockchain-core';
+
+/**
+ * values to handle one single Websocket connection
+ */
+let web3;
+
+/**
+ * Returns a new web3 instances. If a web3 currentProvider is provided, it will
+ * be used.
+ *
+ * @param      {string}  url     url for the web socket connection
+ * @return     {any}     web3 instance
+ */
+export function getWeb3Instance(url = config.web3Provider): any {
+  try {
+    if (!web3) {
+      const provider = new (Web3 as any).providers.WebsocketProvider(
```

Comment:
There seems to be a big discrepancy between the types from api-bbc and what we use in our code. I can't find anything on `WebsocketProvider` in the whole web3 folder 😅

Reply:
https://web3js.readthedocs.io/en/v2.0.0-alpha/web3.html#providers